cmd/clusterctl/hack/create-local-repository.py

```diff
@@ -102,26 +103,28 @@ def load_providers():
         except  Exception as e:
             raise Exception('failed to load clusterctl-settings.json from repo {}: {}'.format(repo, e))
 
+
 def execCmd(args):
     try:
         out = subprocess.Popen(args,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.STDOUT)
+                               stdout=subprocess.PIPE,
+                               stderr=subprocess.STDOUT)
 
         stdout, stderr = out.communicate()
         if stderr is not None:
             raise Exception('stderr contains: \n{}'.format(stderr))
 
         return stdout
-    except  Exception as e:
+    except Exception as e:
         raise Exception('failed to run {}: {}'.format(args, e))
 
-def get_home():
-    return os.path.expanduser('~')
 
 def get_repository_folder():
-    home = get_home()
-    return os.path.join(home, '.cluster-api', 'dev-repository')
+    config_dir = os.getenv("XDG_CONFIG_HOME")
+    if config_dir is None:
+        raise Exception('XDG_CONFIG_HOME variable is not set')
```

Comment:
IMO this should probably be defaulted - according to the spec: https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html

` $XDG_CONFIG_HOME defines the base directory relative to which user-specific configuration files should be stored. If $XDG_CONFIG_HOME is either not set or empty, a default equal to $HOME/.config should be used. `